### PR TITLE
Rework firestarter and extended firestarter

### PR DESCRIPTION
**Pull request by Coolthulhu**
- Firestarter and extended firestarter merged into one actor
- Whether the action will be extended depends on move cost - it becomes extended if it would cost 200+ moves
- LENS flag obsoleted, is implied by actor parameters now
- Good fuel affects time to start the fire: gasoline or oil will make it start near instantly, paper will halve the time required
- For bad fuels (wood, cotton), the time is increased significantly: starting a shrub/corpse fire with a lighter will take 3 to 5 turns, depending on skill level
- Removed the cost multiplier on fire drills. It always takes `charges_per_use` now. Camp fire drills become infinite, because 1000 uses is infinite enough.

### CODE REVIEW

**Comment by mugling** [comment]
> Camp fire drills become infinite, because 1000 uses is infinite enough.

Could they not be `charges_per_use` zero?

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -1152,32 +1152,23 @@ void activity_handlers::start_fire_finish( player_activity *act, player *p )
...
-        float progress_left = float(act->moves_left) /
-                              float(actor->calculate_time_for_lens_fire(p, previous_natural_light_level));
-        act->moves_left = int(progress_left *
-                              (actor->calculate_time_for_lens_fire(p, natural_light_level)));
+    p->mod_moves( -p->moves );
+    const auto actor = dynamic_cast<const firestarter_actor*>( usef->get_actor_ptr() );
+    float light = actor->light_mod( p->pos() );
+    act->moves_left -= light * 100;
+    if( light < 0.1 ) {
+        add_msg( m_bad, _("There is not enough sunlight to start a fire now. You stop trying.") );
```
Comment (mugling) [inline src/activity_handlers.cpp]: Better worded as *There is **now** not enough sunlight to start a fire **so** you stop trying*

File: src/item.h [hunk 2]
```diff
@@ -636,8 +636,9 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
     /**
      * Whether the items is flammable. (Make sure to keep this in sync with
      * fire code in fields.cpp)
+     * @param threshold Item is flammable if it provides more fuel than threshold.
```
Comment (mugling) [inline src/item.h]: More info need about `threshold`. Where is it read from? What are typical values etc

File: src/iuse_actor.cpp [hunk 3]
```diff
@@ -937,122 +940,89 @@ void firestarter_actor::resolve_firestarter_use( const player *p, const item *,
...
+            _("If the current weather holds, it will take around %d minutes to light a fire.");
+        static const std::string normal_msg =
+            _("At your skill level, it will take around %d minutes to light a fire.");
+        p->add_msg_if_player( m_info, ( need_sunlight ? sun_msg : normal_msg ).c_str(),
+            moves / MOVES( MINUTES( 1 ) ) );
+    } else if( moves < MOVES( 2 ) ) {
+        // If less than 2 turns, don't start a long action
+        resolve_firestarter_use( p, it, pos );
+        p->mod_moves( -moves );
+        return it->type->charges_to_use();
```
Comment (mugling) [inline src/iuse_actor.cpp]: If you're using `ammo_capacity()` and `ammo_remaining()` you may as well use `ammo_required()`

**Comment by Rivet-the-Zombie** [comment]
Rainy/wet weather should really make it harder to do/take longer.

**Comment by mugling** [comment]
> Rainy/wet weather should really make it harder to do/take longer.

This could make survival much more relevant. Looking at the above code it should be possible to occasionally 'miss' a turn in bad weather?

**Comment by nikheizen** [comment]
1 minute wouldn't be too short for lighting a shrub or corpse on fire.
Would remove the tech of fire-cheesing tough monsters while still being short enough to be irrelevant to crafting/cooking.
Unless you picked 3-5 to keep it tactically relevant.

**Pull request merged**